Sources/ComposableArchitecture/SwiftUI/Binding.swift

```diff
@@ -1,306 +1,303 @@
+  public init(from decoder: Decoder) throws {
+    do {
+      let container = try decoder.singleValueContainer()
+      self.init(wrappedValue: try container.decode(Value.self))
+    } catch {
+      self.init(wrappedValue: try Value(from: decoder))
     }
   }
+}
 
-  extension BindableState: Encodable where Value: Encodable {
-    public func encode(to encoder: Encoder) throws {
-      do {
-        var container = encoder.singleValueContainer()
-        try container.encode(self.wrappedValue)
-      } catch {
-        try self.wrappedValue.encode(to: encoder)
-      }
+extension BindableState: Encodable where Value: Encodable {
+  public func encode(to encoder: Encoder) throws {
+    do {
+      var container = encoder.singleValueContainer()
+      try container.encode(self.wrappedValue)
+    } catch {
+      try self.wrappedValue.encode(to: encoder)
     }
   }
+}
 
-  extension BindableState: CustomReflectable {
-    public var customMirror: Mirror {
-      Mirror(reflecting: self.wrappedValue)
-    }
+extension BindableState: CustomReflectable {
+  public var customMirror: Mirror {
+    Mirror(reflecting: self.wrappedValue)
   }
+}
 
-  extension BindableState: CustomDumpRepresentable {
-    public var customDumpValue: Any {
-      self.wrappedValue
-    }
+extension BindableState: CustomDumpRepresentable {
+  public var customDumpValue: Any {
+    self.wrappedValue
   }
+}
 
-  extension BindableState: CustomDebugStringConvertible where Value: CustomDebugStringConvertible {
-    public var debugDescription: String {
-      self.wrappedValue.debugDescription
-    }
+extension BindableState: CustomDebugStringConvertible where Value: CustomDebugStringConvertible {
+  public var debugDescription: String {
+    self.wrappedValue.debugDescription
   }
+}
 
-  /// An action type that exposes a `binding` case that holds a ``BindingAction``.
-  ///
-  /// Used in conjunction with ``BindableState`` to safely eliminate the boilerplate typically
-  /// associated with mutating multiple fields in state.
-  ///
-  /// See the documentation for ``BindableState`` for more details.
-  public protocol BindableAction {
-    /// The root state type that contains bindable fields.
-    associatedtype State
+/// An action type that exposes a `binding` case that holds a ``BindingAction``.
+///
+/// Used in conjunction with ``BindableState`` to safely eliminate the boilerplate typically
+/// associated with mutating multiple fields in state.
+///
+/// See the documentation for ``BindableState`` for more details.
+public protocol BindableAction {
+  /// The root state type that contains bindable fields.
+  associatedtype State
 
-    /// Embeds a binding action in this action type.
-    ///
-    /// - Returns: A binding action.
-    static func binding(_ action: BindingAction<State>) -> Self
-  }
+  /// Embeds a binding action in this action type.
+  ///
+  /// - Returns: A binding action.
+  static func binding(_ action: BindingAction<State>) -> Self
```

Comment:
I know not your code, but this is some dark black magic. Does this work because an enum case constructor is a static function on the enum's type and so technically fulfills this requirement?